aten/src/ATen/GeneratorForPrivareuseone.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <c10/util/intrusive_ptr.h>
+
+namespace at {
+
+using GeneratorFuncType = std::function<at::Generator(c10::DeviceIndex)>;
+
+static std::map<std::string, GeneratorFuncType>& GetGeneratorPrivateClassMap() {
+  static std::map<std::string, GeneratorFuncType> generator_privateuse1_map;
+  return generator_privateuse1_map;
```

Comment:
Can you make this a `c10::optional` instead of a map, since it only ever gets one entry?